engine/time/src/main/java/io/deephaven/time/calendar/CalendarDay.java

```diff
@@ -63,7 +64,8 @@ public class CalendarDay<T extends Comparable<T> & Temporal> {
             }
         }
 
-        this.businessTimeRanges = List.of(ranges);
+        this.businessTimeRanges = ranges;
+        this.businessNanos = Arrays.stream(businessTimeRanges).map(TimeRange::nanos).reduce(0L, Long::sum);
```

Comment:
Missed this earlier; but you can create a long stream and call sum directly: `.mapToLong(TimeRange::nanos).sum()`.